firmware/shared/code/I2C.c

```diff
@@ -0,0 +1,268 @@
+/*
+ * I2C.c
+ *
+ *  Created on: Oct 10, 2020
+ *      Author: Shuhang Tan
+ *      Reference: https://github.com/uvic-auvic/Firmware-Motor_Controller_2.0/blob/master/src/I2C.c
+ *                 (I2C code for motor controller 2)
+ */
+
+#include "I2C.h"
+#include <string.h>
+#include <stdlib.h>
+#include "stm32f4xx.h"
+#include "LED.h"
+
+// I2C SDA and SCL pins
+#define _I2C_SDA_GPIO  (GPIO_Pin_9) // PC9
+#define _I2C_SCL_GPIO  (GPIO_Pin_8) // PA8
+
+// I2C sendBuffer size
+#define _I2C_BUFFER_SIZE    (10U)
+
+typedef struct
+{
+	uint8_t Buffer[_I2C_BUFFER_SIZE];
+} I2C_data_S;
+
+// Adding an idle state because reading the BUSY bit in SR2 can cause problems
+typedef enum{
+	I2C_STATE_SEND,    // Write
+	I2C_STATE_RECEIVE, // Read
+	I2C_STATE_IDLE,    // Idle
+}I2C_state_S;
+
+// Global variables and their initialization
+static I2C_data_S I2C_send_data;

```

Comment:
There is an `I2C_data_S` struct above to put static variables in.

Reply:
Most of the global variables are in the struct now